compose/project.py

```diff
@@ -274,6 +274,9 @@ def remove_stopped(self, service_names=None, **options):
             service.remove_stopped(**options)
 
     def containers(self, service_names=None, stopped=False, one_off=False):
+        if service_names:
+            # Will raise NoSuchService if one of the names is invalid
+            self.get_services(service_names)
```

Comment:
Yeah, I made a comment along those lines in the pull request itself - it would be very easy to add a `validate_service_names` method that *just* makes sure the services actually exist, if you'd prefer.